authorization/invitation/service/invitation_service.go

```diff
@@ -0,0 +1,154 @@
+// This method creates one record in the INVITATION table for each user in the invitations parameter.  Any roles that are issued
+// as part of a user's invitation are created in the INVITATION_ROLE table.
+func (s *InvitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UUID, inviteTo string, invitations []invitation.Invitation) error {
+	var inviteToIdentity *account.Identity
+	var identityResource *resourceRepo.Resource
+	var inviteToResource *resourceRepo.Resource
+
+	// First try to convert inviteTo to a uuid
+	inviteToUUID, err := uuid.FromString(inviteTo)
+	// If we get an error here, the value is definitely not for an Identity so we'll treat it as a resource ID
+	if err != nil {
+		// Try to lookup a resource with the same ID value
+		inviteToResource, err = s.repo.ResourceRepository().Load(ctx, inviteTo)
+		if err != nil {
+			return errors.NewNotFoundError(fmt.Sprintf("invalid identifier '%s' provided for organization, team, security group or resource", inviteTo), inviteTo)
+		}
+	}
+
+	// If we didn't successfully find a valid resource already, it means the inviteTo is a UUID
+	if inviteToResource == nil {
+		// Attempt to lookup the identity of the organization, team or security group that invitations will be issued for
+		inviteToIdentity, err = s.repo.Identities().Load(ctx, inviteToUUID)
+		if err != nil {
+			// That didn't work, try to lookup a resource with the same ID value
+			inviteToResource, err = s.repo.ResourceRepository().Load(ctx, inviteTo)
+			if err != nil {
+				return errors.NewNotFoundError(fmt.Sprintf("invalid identifier '%s' provided for organization, team, security group or resource", inviteTo), inviteTo)
+			}
+		}
+	}
+
+	if inviteToIdentity != nil {
+		// Load the resource for the identity
+		identityResource, err = s.repo.ResourceRepository().Load(ctx, *inviteToIdentity.IdentityResourceID)
+		if err != nil {
+			return errors.NewInternalErrorFromString(ctx, "error loading resource for identity")
+		}
+
+		// Confirm that the issuing user has the necessary scope to manage members for the organization, team or security group
+		scope, err := s.permSvc.HasScope(ctx, issuingUserId, *inviteToIdentity.IdentityResourceID, authorization.ManageMembersScope)
+		if err != nil {
+			return errors.NewInternalError(ctx, err)
+		}
+
+		if !scope {
+			return errors.NewForbiddenError(fmt.Sprintf("user requires %s scope to invite other users", authorization.ManageMembersScope))
+		}
+
+		// We only allow membership in some identity types - confirm that we are inviting to an organization, team or security group
+		if !authorization.CanHaveMembers(identityResource.ResourceType.Name) {
+			return errors.NewInternalErrorFromString(ctx, "may only invite a user to an organization, team or security group")
+		}
+	} else if inviteToResource != nil {
+		// Confirm that the issuing user has the manage members scope for the resource
```

Comment:
This whole case when we invite to a resource is not covered by tests.
There should be two tests for this:
1. User has permissions to invite to a resource
2. User doesn't have permissions.

Reply:
I've added tests to cover both of these